### PR TITLE
eslint: Enable some more rules and fix code accordingly

### PR DESCRIPTION
**Pull request by martinpitt**
Most of these are space fixes, but the last four are some useful code simplifications.

There's certainly more to be done, but let's split this up into a few smaller PRs.

CC @mareklibra

### CODE REVIEW

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx [hunk 1]
```diff
@@ -109,13 +109,13 @@ class CreateVM extends React.Component {
     }
 
     onChangedEventValue(key, e) {
-        if (e && e.target && typeof(e.target.value) !== 'undefined') {
+        if (e && e.target && typeof e.target.value !== 'undefined') {
```
Comment (mareklibra) [inline pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: IMO, this makes code harder to read.
Probably introduced by https://eslint.org/docs/rules/space-unary-ops

File: pkg/machines/components/libvirtSlate.jsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ class LibvirtSlate extends React.Component {
     }
 
     onLibvirtEnabledChanged(e) {
-        if (e && e.target && typeof(e.target.checked) === "boolean") {
+        if (e && e.target && typeof e.target.checked === "boolean") {
```
Comment (mareklibra) [inline pkg/machines/components/libvirtSlate.jsx]: likewise

File: pkg/packagekit/updates.jsx [hunk 3]
```diff
@@ -665,8 +665,8 @@ class OsUpdates extends React.Component {
                     if (role !== PK.Enum.ROLE_UPDATE_PACKAGES)
                         return;
                     // data looks like:
-                    // downloading	bash-completion;1:2.6-1.fc26;noarch;updates-testing
-                    // updating	bash-completion;1:2.6-1.fc26;noarch;updates-testing
+                    // downloading\tbash-completion;1:2.6-1.fc26;noarch;updates-testing
```
Comment (mareklibra) [inline pkg/packagekit/updates.jsx]: With this batch updated (multiple rules at once), it's hard to find which rules introduced that.
I think we should avoid updated inside comments

**Comment by mareklibra** [comment]
In general, these rules seem mostly reasonable to me. Just minor comments regarding two of them.

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: We use `typeof` without the extra parentheses a lot (and it's an operator, so it's syntactically valid). I'm fine with adding them back and instead having `typeof (foo)`, that's also ok.

File: pkg/packagekit/updates.jsx (hunk 3 above)
Comment (martinpitt) [reply pkg/packagekit/updates.jsx]: This is why this PR has one commit per rule. This one is the "no-tabs" one.

File: pkg/machines/components/create-vm-dialog/createVmDialog.jsx (hunk 1 above)
Comment (mareklibra) [reply pkg/machines/components/create-vm-dialog/createVmDialog.jsx]: ok

**Pull request merged**